timescaledb/tutorials/aws-lambda/continuous-deployment.md

```diff
@@ -0,0 +1,180 @@
+    git remote add origin <YOUR_GITHUB_PROJECT_URL.git>
+    git push -u origin master
+    ```
+
+At this point, you have a GitHub repo with just the Lambda function in it. Let's see how to connect this repo to the AWS Lambda function.
+
+## Connect GitHub and AWS Lambda
+
+### Procedure: Adding your AWS credentials to the repository
+You need to add your AWS credentials to the repository so it will have permission to connect to Lambda. You can do this
+by adding [GitHub secrets](https://docs.github.com/en/actions/reference/encrypted-secrets) using the GitHub CLI.
+
+1.  Authenticate with a GitHub:
+    ```bash
+    gh auth login
+    ```
+    This will prompt you to choose what account you want to log into using either your password or GitHub
+    authentication token.
+1.  Add AWS credentials as GitHub secrets. 
+    By using GitHub secrets, your credentials are encrypted and cannot be seen 
+    publicly. Use the `gh secret set` command to upload your AWS credentials one by one 
+    (you'll be prompted to paste the values for each one):
+    
+    AWS_ACCESS_KEY_ID:
+    ```bash
+    gh secret set AWS_ACCESS_KEY_ID
+    ```
+    
+    AWS_SECRET_ACCESS_KEY:
+    ```bash
+    gh secret set AWS_SECRET_ACCESS_KEY
+    ```
+    
+    AWS_REGION:
+    ```bash
+    gh secret set AWS_REGION
+    ```
+    
+    To make sure your credentials have been uploaded correctly, you can list the available GitHub secrets:
+    ```bash
+    gh secret list
+    AWS_ACCESS_KEY_ID      Updated 2021-09-13
+    AWS_SECRET_ACCESS_KEY  Updated 2021-09-13
+    AWS_REGION             Updated 2021-09-13
+    ```
+
+Now you know that you have your AWS credentials available for the repository to use.
+
+### Procedure: Setting up GitHub actions
+You will set up a new automation based on the ["AWS Lambda Deploy" GitHub action](https://github.com/marketplace/actions/aws-lambda-deploy)
+to auto-deploy to AWS Lambda.
+
+1.  Create a new YAML file (this will contain the deployment configuration):
+    ```bash
+    touch .github/workflows/main.yml
+    ```
+    
+    And add this to the file:
+    ```yml
+    name: deploy to lambda
+    on:
+      # Trigger the workflow on push or pull request,
+      # but only for the main branch
+      push:
+        branches:
+          - master
+    jobs:
+    
+      deploy_source:
+        name: deploy lambda from source
+        runs-on: ubuntu-latest
+        steps:
+          - name: checkout source code
+            uses: actions/checkout@v1
+          - name: default deploy
+            uses: appleboy/lambda-action@master
+            with:
+              aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+              aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+              aws_region: ${{ secrets.AWS_REGION }}
+              function_name: lambda-cd
+              source: function.py
+    ```
+    This configuration will make sure to deploy the code to Lambda when there's a new push to the master branch.
```

Comment:
```suggestion
    This configuration ensures that the code is deployed to Lambda when there's a new push to the `master` branch.
```

Shouldn't we be using `main` now?

Reply:
Does it make a difference? I've been using master throughout each AWS tutorial, so I kept the same here. Maybe we should do a search and replace on each Lambda tutorial to master->main?

Reply:
GH have updated the default to `main`, in a general move away from language that could be considered insensitive. See also https://inclusivenaming.org/. I think we probably should do a search through, yes.

Reply:
to be fixed here https://github.com/timescale/docs/issues/456